### PR TITLE
Prepare Azure Core for April 2020 Release

### PR DESCRIPTION
**Pull request by alzimmermsft**
This PR prepares Azure Core for the April 2020 release.

### CODE REVIEW

File: sdk/core/azure-core-http-netty/CHANGELOG.md [hunk 1]
```diff
@@ -2,6 +2,8 @@
 
 ## 1.5.0-beta.1 (Unreleased)
```
Comment (srnagar) [inline sdk/core/azure-core-http-netty/CHANGELOG.md:3]: Should the version be updated here to 1.5.0 with a release date?

File: sdk/core/azure-core/CHANGELOG.md [hunk 2]
```diff
@@ -2,7 +2,14 @@
 
 ## 1.4.0-beta.1 (Unreleased)
```
Comment (srnagar) [inline sdk/core/azure-core/CHANGELOG.md:3]: Version update with release date.

File: sdk/core/azure-core-http-okhttp/CHANGELOG.md [hunk 3]
```diff
@@ -2,6 +2,7 @@
 
 ## 1.3.0-beta.1 (Unreleased)
```
Comment (srnagar) [inline sdk/core/azure-core-http-okhttp/CHANGELOG.md:3]: Version update and release date.

File: sdk/core/azure-core-tracing-opentelemetry/pom.xml [hunk 4]
```diff
@@ -40,7 +40,7 @@
     <dependency>
       <groupId>com.azure</groupId>
       <artifactId>azure-core</artifactId>
-      <version>1.4.0-beta.1</version> <!-- {x-version-update;com.azure:azure-core;current} -->
```
Comment (samvaity) [inline sdk/core/azure-core-tracing-opentelemetry/pom.xml:43]: Could you also update the changelog for tracing to have the release date?

File: eng/versioning/version_client.txt [hunk 5]
```diff
@@ -43,8 +43,8 @@ com.azure:perf-test-core;1.0.0-beta.1;1.0.0-beta.1
 # Format;
 # unreleased_<groupId>:<artifactId>;dependency-version
 # note: The unreleased dependencies will not be manipulated with the automatic PR creation code.
-unreleased_com.azure:azure-core;1.4.0-beta.1
-unreleased_com.azure:azure-core-amqp;1.1.0-beta.1
+unreleased_com.azure:azure-core;1.4.0
+unreleased_com.azure:azure-core-amqp;1.0.2
 unreleased_com.azure:azure-core-test;1.2.0-beta.1
```
Comment (JimSuplizio) [inline eng/versioning/version_client.txt:48]: I'm curious, should we also be releasing azure-core-test? The last time it was released was in [January](https://search.maven.org/artifact/com.azure/azure-core-test). Right now, we've got azure-data-appconfiguration, azure-security-keyvault-certificates, azure-security-keyvault-keys and azure-security-keyvault-secrets dependent on this unreleased version (test only dependencies though).

Comment (alzimmermsft) [reply eng/versioning/version_client.txt]: Updated to release 1.2.0

**Pull request merged**